addons/resource/models/resource.py

```diff
@@ -354,11 +351,11 @@ def plan_days(self, days, day_dt, compute_leaves=False, domain=None):
 
         # which method to use for retrieving intervals
         if compute_leaves:
-            get_intervals = partial(self._work_intervals, domain=domain)
+            get_intervals = partial(self._work_intervals, domain=domain, resource=resource)
         else:
             get_intervals = self._attendance_intervals
 
-        if days > 0:
+        if days >= 0:
```

Comment:
The logic for 0 days does not work here.
You need specific code for the case `days == 0`.
```python
# find the first interval, and return its start datetime
delta = timedelta(days=14)
for n in range(100):
    dt = day_dt + delta * n
    for start, stop, meta in get_intervals(dt, dt + delta):
        return revert(start)
return False
```